### ISSUE TITLE
Add host does not work as expected when a ssh://host _and_ the username field are both used

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#17635]
### Explain what happens

1. Visit the host switcher
2. Add a new host
3. In the "Add a new host" dialog, use an SSH URI without a user, such as ssh://host
4. Put your username in the username field
5. Try to connect... it won't work

Screenshot:

![image](https://user-images.githubusercontent.com/10246/183675712-d3b51ba1-630b-44e4-91a3-c1c5413d6a16.png)

It then shows this for a while:

![Screenshot from 2022-08-09 16-32-02](https://user-images.githubusercontent.com/10246/183676137-fbb621a4-675f-4f23-8a6a-1d9cf3b6d988.png)

And finally this error message:

![Screenshot from 2022-08-09 16-32-11](https://user-images.githubusercontent.com/10246/183676198-bf4bc91a-17ae-4f31-897e-c92b2f8766aa.png)

(Also: In this case, the "Cancel" link should really be a "Close" button with a secondary style, not a link styled button.)

Please note that the host + user works, as does the ssh://user@host... but mixing the URI + user doesn't.

In other words: Cockpit should try `ssh://user@host` (with the username from the separate entry) when `ssh://host` is used and the user is also filled in.

### Version of Cockpit

273

### Where is the problem in Cockpit?

Navigation & Shell

### Server operating system

CentOS

### Server operating system version

Stream 8

### What browsers are you using?

Firefox, Chrome
